db/data_migration/20150730134022_republish_routes_for_all_detailed_guides.rb

```diff
@@ -0,0 +1,21 @@
+require 'plek'
+require 'gds_api/router'
+
+router_api = GdsApi::Router.new(Plek.find('router-api'))
+
+scope = Document.distinct.
+  joins(:editions).
+  where(document_type: "DetailedGuide",
+    editions: {state: Edition::POST_PUBLICATION_STATES})
+
+count = scope.count
+
+scope.each_with_index do |guide, i|
+  if guide.ever_published_editions.any?
+    slug = guide.slug.sub(%r{^deleted-}, '')
+    puts "Adding route #{i+1}/#{count} for /#{slug}"
+    router_api.add_route("/#{slug}", "exact", "whitehall-frontend")
```

Comment:
We shouldn't be doing this directly to the router.  This has to go via publishing-api so that it gets checked by the url-arbiter.  There are some detailed guides that conflict with mainstream content (one of them is at `/foreign-travel-advice` which is particularly bad), and this will clobber their routes.